closed/src/java.base/share/classes/sun/security/ec/NativeECDSASignature.java

```diff
@@ -541,8 +541,14 @@ protected byte[] engineSign() throws SignatureException {
         int sigLen = ((params.getOrder().bitLength() + 7) / 8) * 2;
         byte[] sig = new byte[sigLen];
 
-        ECDSAOperations.forParameters(params)
-                .orElseThrow(() -> new SignatureException("Curve not supported: " + params));
+        try{
+            ECDSAOperations.forParameters(params)
+                    .orElseThrow(() -> new SignatureException("Curve not supported: " + params));
+        } catch (SignatureException se) {
+            if (!(ECUtil.isBrainpoolP512r1(params))) {
+                throw se;
+            }
+        }
```

Comment:
Again, don't throw an exception that will just be suppressed:
```java
        if (ECDSAOperations.forParameters(params).isEmpty()
                && !NativeECUtil.isBrainpoolP512r1(params)
        ) {
             throw new SignatureException("Curve not supported: " + params);
        }
```

Reply:
Updated as suggested to avoid using exception handling for flow control.